programs/drift/src/math/margin.rs

```diff
@@ -576,12 +578,28 @@ pub fn meets_place_order_margin_requirement(
         return Err(ErrorCode::InsufficientCollateral);
     }
 
-    if calculation.get_num_of_liabilities()? > 1 {
-        validate!(
-            !calculation.with_isolated_liability,
-            ErrorCode::IsolatedAssetTierViolation,
-            "User attempting to increase number of liabilities above 1 with a isolated tier liability"
-        )?;
+    if calculation.with_isolated_liability {
+        if !user.is_reduce_only() {
+            validate!(
+                calculation.num_perp_liabilities <= 1,
+                ErrorCode::IsolatedAssetTierViolation,
+                "User attempting to increase perp liabilities above 1 with a isolated tier liability"
+            )?;
+
+            validate!(
+                !user.has_spot_bid_order()?,
```

Comment:
should just force margin trading be disabled? you can have a spot bid that doesn't create borrow?

Reply:
forcing margin_trading_disabled